README.md

```diff
@@ -1,35 +1,351 @@
+
+### Conditions
+
+Conditions are a list of conditions specified inline. Each condition has a detector struct within and two extra parameters of minNumFindings and minConfidence as below.
+
+```json
+{
+  "conditions": [
+    {
+      "minNumFindings": 1,
+      "minConfidence": "POSSIBLE",
+      "detector": {}
+    }
+  ]
+}
+```
+
+minNumFindings specifies minimal number of findings to trigger findings to return for one request, e.g. if you set minNumFindings to be 2, and only 1 finding within the request payload, then this finding will be filtered.
+
+minConfidence specifies the minimal threshold to trigger a potential finding to be captured, in total we have five levels of confidence from least sensitive to most sensitive:
+
+- VERY_LIKELY
+- LIKELY
+- POSSIBLE
+- UNLIKELY
+- VERY_UNLIKELY
+
+### Detector
+
+A detector is either a prebuilt detector from nightfall or customized regex | wordlist by customer, specified by
+detectorType field.
+
+- nightfall prebuilt detector
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "NIGHTFALL_DETECTOR",
+      "nightfallDetector": "API_KEY",
+      "displayName": "credit card detector"
+    }
+  }
+  ```
+
+  Within detector struct
+
+  - First specify detectorType as NIGHTFALL_DETECTOR
+  - Pick the nightfall detector you want from the list
+    - API_KEY
+    - CRYPTOGRAPHIC_KEY
+    - RANDOMLY_GENERATED_TOKEN
+    - CREDIT_CARD_NUMBER
+    - US_SOCIAL_SECURITY_NUMBER
+    - AMERICAN_BANKERS_CUSIP_ID
+    - US_BANK_ROUTING_MICR
+    - ICD9_CODE
+    - ICD10_CODE
+    - US_DRIVERS_LICENSE_NUMBER
+    - US_PASSPORT
+    - PHONE_NUMBER
+    - IP_ADDRESS
+    - EMAIL_ADDRESS
+  - Put a display name for your detector, as this will be attached on your findings
+
+- customized regex
+
+  For convenience, we also support customized regex as a detector, which you can fill in as follow.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "REGEX",
+      "regex": {
+        "pattern": "coupon",
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleRegexCouponDetector"
+    }
+  }
+  ```
+
+- word list
+
+  In case you are not famaliar what regex is, we also support word list searching, simply put in a list of words you want to find.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "WORD_LIST",
+      "wordList": {
+        "values": ["key", "credential"],
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleWordListKeyDetector"
+    }
+  }
+  ```
+
+- [Extra Parameters Within Detector]
+
+  Besides to define which detector to call, we also allow you to optionally specify some rules you want to attach surround the findinds and finding themselves. They are contextRules and exclusionRules.
+
+  - contextRules
+    A context rule is defined as the surrounding context(pre/post chars) of a finding, you can define a trigger/rule in this section. Once it's triggered, the confidence of finding will be adjusted accordingly base on your confid.
```

Comment:
```suggestion
    A context rule evaluates the surrounding context(pre/post chars) of a finding and adjusts the finding's confidence if the input context rule pattern exists.
```

Reply:
A context rule evaluates the surrounding context(pre/post chars) of a finding. Once it's triggered, the confidence of finding will be adjusted accordingly base on your config.